### ISSUE TITLE
`arm64` step of docker end user workflow is failing without a meaningful log

### ISSUE THREAD

**Issue by francesco-ballarin** [FEniCS/dolfinx#2846]
While testing #2844, both runs in https://github.com/FEniCS/dolfinx/actions/runs/6717503005/job/18255469197 and https://github.com/FEniCS/dolfinx/actions/runs/6717953807/job/18256739847 failed on `arm64` while building the `pybind11`.  In both cases, the output shows
```
#17 36.84   *** Building project with Ninja...
#17 41.33   [1/13] Building CXX object CMakeFiles/cpp.dir/dolfinx/wrappers/dolfinx.cpp.o
#17 43.42   [2/13] Building CXX object CMakeFiles/cpp.dir/dolfinx/wrappers/graph.cpp.o
#17 46.45   [3/13] Building CXX object CMakeFiles/cpp.dir/dolfinx/wrappers/common.cpp.o
#17 47.55   [4/13] Building CXX object CMakeFiles/cpp.dir/dolfinx/wrappers/geometry.cpp.o
#17 52.78   [5/13] Building CXX object CMakeFiles/cpp.dir/dolfinx/wrappers/log.cpp.o
```
and no furthe errors. In #2844 I imagined that we were out of build minutes, but after seeing the same failure twice at exactly the same point, my suspicion would be `/tmp` runs out of space. I don't have access to a `arm64` CPU so I cannot test the workflow locally.

**Comment by jhale** [comment]
I am getting a crash at a different time. The instances have 64GB of SSD so /tmp is not an issue. Possibly a problem at BuildJet's end, will wait a day and see if it sorts itself out.

https://github.com/FEniCS/dolfinx/actions/runs/6732618747

**Comment by jhale** [comment]
No issues building locally.

**Comment by francesco-ballarin** [comment]
> I am getting a crash at a different time.

To me, 
https://github.com/FEniCS/dolfinx/actions/runs/6732618747/job/18299658286
looks like the same point as in the opening post
https://github.com/FEniCS/dolfinx/actions/runs/6717953807/job/18256739847

**Comment by jhale** [comment]
It's an out-of-memory error. The old build system was using make, the new one ninja, which uses more build threads.

Fixed in #2860

**Comment by jhale** [comment]
It's an out-of-memory error. The old build system was using make, the new one ninja, which uses more build threads.

Fixed in #2860.
